Automattic/jetpack
# tools: Default some prompts to "yes"

## Proposed changes:

Some of the prompts, such as the one about pushing the release branch after creating it, really should default to "yes" so if someone is just hitting enter they do the thing that's right most of the time.

So, add an option to the `proceed_p` shell function to specify that the default should be "yes", and use that in various places where it seems to make sense.

### Other information:


## Jetpack product discussion


p1716905006338979/1716899514.277269-slack-C01U2KGS2PQ

## Does this pull request change what data or activity we track or use?


no

## Testing instructions:






* In a bash shell, run `source tools/includes/proceed_p.sh` then try out the following with various answers:
  * A prompt defaulting to no: `if proceed_p ""; then echo "answered yes"; else echo "answered no"; fi`
  * A prompt defaulting to yes: `if proceed_p "" "" true; then echo "answered yes"; else echo "answered no"; fi`
* Check that the changed calls in various tools make sense to default to "yes".
* Also double check that the added `true` is always being passed as the third argument, not accidentally the second.
